src/charm.py

```diff
@@ -178,6 +169,26 @@ def _configure_amf(self, _: EventBase) -> None:
         except ValueError:
             return
 
+    def _store_certificate_and_key(self) -> bool:
+        """Store certificate and private key in workload.
+
+        Returns:
+            bool: True if certificate or private key is updated
+        """
```

Comment:
I do not have a better idea in mind at the moment, but the name of the method and what it returns feels wrong. When I read the line above that uses this, I was confused and had to come down here to read the code.

Reply:
I tried to improve the function name and docstring, let me know if that's better.